Cubical/Algebra/AbGroup/Instances/Hom.agda

```diff
@@ -0,0 +1,147 @@
+  zero .snd .pres1 = refl
+  zero .snd .presinv a = sym (inv0B)
+
+
+  -- Pointwise addition of morphisms
+  module _ (f* g* : AbGroupHom A B) where
+    private
+      f = f* .fst
+      g = g* .fst
+
+    HomAdd : AbGroupHom A B
+    HomAdd .fst = λ a → f a + g a
+
+    HomAdd .snd .pres· a a' =
+        f (a ⋆ a') + g (a ⋆ a')           ≡⟨ cong (_+ g(a ⋆ a')) (f* .snd .pres· _ _) ⟩
+        (f a + f a') + g (a ⋆ a')         ≡⟨ cong ((f a + f a') +_) (g* .snd .pres· _ _) ⟩
+        (f a + f a') + (g a + g a')       ≡⟨ sym (assoc _ _ _) ⟩
+        f a + (f a' + (g a + g a'))       ≡⟨ cong (f a +_) (assoc _ _ _) ⟩
+        f a + ((f a' + g a) + g a')       ≡⟨ cong (λ b → (f a + b + g a')) (comm _ _) ⟩
+        f a + ((g a + f a') + g a')       ≡⟨ cong (f a +_) (sym (assoc _ _ _)) ⟩
+        f a + (g a + (f a' + g a'))       ≡⟨ assoc _ _ _ ⟩
+        (f a + g a) + (f a' + g a')       ∎
+
+    HomAdd .snd .pres1 =
+        f 0A + g 0A       ≡⟨ cong (_+ g 0A) (hom0AB f*) ⟩
+        0B + g 0A         ≡⟨ cong (0B +_) (hom0AB g*) ⟩
+        0B + 0B           ≡⟨ idrB _ ⟩
+        0B                ∎
+
+    HomAdd .snd .presinv a =
+        f (inv a) + g (inv a)     ≡⟨ cong (_+ g (inv a)) (homInvAB f* _) ⟩
+        (- f a) + g (inv a)       ≡⟨ cong ((- f a) +_) (homInvAB g* _) ⟩
+        (- f a) + (- g a)         ≡⟨ comm _ _ ⟩
+        (- g a) + (- f a)         ≡⟨ sym (invDistr _ _) ⟩
+        - (f a + g a)             ∎
+
+
+  -- Pointwise inverse of morphism
+  module _ (f* : AbGroupHom A B) where
+    private
+      f = f* .fst
+
+    HomInv : AbGroupHom A B
+    HomInv .fst = λ a → - f a
+
+    HomInv .snd .pres· a a' =
+        - f (a ⋆ a')            ≡⟨ cong -_ (f* .snd .pres· _ _) ⟩
+        - (f a + f a')          ≡⟨ invDistr _ _ ⟩
+        (- f a') + (- f a)      ≡⟨ comm _ _ ⟩
+        (- f a) + (- f a')      ∎
+
+    HomInv .snd .pres1 =
+        - (f 0A)      ≡⟨ cong -_ (f* .snd .pres1) ⟩
+        - 0B          ≡⟨ inv0B ⟩
+        0B            ∎
+
+    HomInv .snd .presinv a =
+        - f (inv a)     ≡⟨ cong -_ (homInvAB f* _) ⟩
+        - (- f a)       ∎
+
+
+  -- Group laws for morphisms
+  private
+    0ₕ = zero
+    _+ₕ_ = HomAdd
+    -ₕ_ = HomInv
+
+  -- Morphism addition is associative
+  HomAdd-assoc : (f g h : AbGroupHom A B) → (f +ₕ (g +ₕ h)) ≡ ((f +ₕ g) +ₕ h)
+  HomAdd-assoc f g h = GroupHom≡ (funExt λ a → assoc _ _ _)
+
+  -- Morphism addition is commutative
+  HomAdd-comm : (f g : AbGroupHom A B) → (f +ₕ g) ≡ (g +ₕ f)
+  HomAdd-comm f g = GroupHom≡ (funExt λ a → comm _ _)
+
+  -- zero is right identity
+  HomAdd-zero : (f : AbGroupHom A B) → (f +ₕ zero) ≡ f
+  HomAdd-zero f = GroupHom≡ (funExt λ a → idrB _)
+
+  -- -ₕ is right inverse
+  HomInv-invr : (f : AbGroupHom A B) → (f +ₕ (-ₕ f)) ≡ zero
+  HomInv-invr f = GroupHom≡ (funExt λ a → invrB _)
+
+
+-- Abelian group structure on AbGroupHom A B
+HomAbStr : (A : AbGroup ℓ) → (B : AbGroup ℓ') → AbGroupStr (AbGroupHom A B)
+HomAbStr A B = abgroupstr
```

Comment:
Please define using copatterns (I think `makeIsAbGroup` can still be used here, so only the top-level need copatterns)